cluster/network.go

```diff
@@ -259,19 +269,71 @@ func (c *Cluster) checkKubeAPIPort(ctx context.Context) error {
 	return nil
 }
 
-func (c *Cluster) deployTCPPortListeners(ctx context.Context) error {
+func (c *Cluster) deployTCPPortListeners(ctx context.Context, currentCluster *Cluster) error {
 	log.Infof(ctx, "[network] Deploying port listener containers")
 	var errgrp errgroup.Group
 
-	portList := []string{
-		KubeAPIPort,
+	etcdHosts := []*hosts.Host{}
+	cpHosts := []*hosts.Host{}
+	workerHosts := []*hosts.Host{}
+	if currentCluster != nil {
+		etcdHosts = hosts.GetToAddHosts(currentCluster.EtcdHosts, c.EtcdHosts)
+		cpHosts = hosts.GetToAddHosts(currentCluster.ControlPlaneHosts, c.ControlPlaneHosts)
+		workerHosts = hosts.GetToAddHosts(currentCluster.WorkerHosts, c.WorkerHosts)
+	} else {
+		etcdHosts = c.EtcdHosts
+		cpHosts = c.ControlPlaneHosts
+		workerHosts = c.WorkerHosts
+	}
+	// deploy ectd listeners
+	etcdPortList := []string{
 		EtcdPort1,
 		EtcdPort2,
-		ScedulerPort,
-		ControllerPort,
+	}
+	for _, host := range etcdHosts {
+		runHost := host
+		errgrp.Go(func() error {
+			return c.deployListener(ctx, etcdPortList, runHost, EtcdPortListenContainer)
+		})
+	}
+	if err := errgrp.Wait(); err != nil {
+		return err
+	}
+
+	// deploy controlplane listeners
+	controlPlanePortList := []string{
+		KubeAPIPort,
+	}
+	for _, host := range cpHosts {
```

Comment:
the code blocks for deploying portlisten containers look pretty similar. Can we extract it to its own function, and just pass list of hosts/container to deploy as an argument?